domain/domain.go

```diff
@@ -0,0 +1,35 @@
+package domain
+
+import (
+	"errors"
+	"net"
+	"net/url"
+	"strings"
+)
+
+var (
+	errInvalidURL = errors.New("invalid URL")
+)
+
+func extractDomain(inputURL string) (string, error) {
```

Comment:
Good luck using that private function from somewhere else 😁

Reply:
🤓